ONSdigital/census-rm-sample-loader
# Load census sample file 0.3 spec, general refactor

## Motivation and Context
We're starting to make changes to this sample loader for new census features so this seemed like a good time to do some refactoring and add some tests. The actual feature change is the move from using 'TLA' and 'REFERENCE' for the sample unit reference to using the 'ARID' code (Note: will not be the final spec for sample unit ref but neither TLA nor REFERENCE are on this 0.3 sample spec so we need a suitable placeholder).

## What has Changed
- Move Rabbit interaction out into rabbit context class
- Move Redis interaction out into redis context class
- Separate sample loader code from the script which calls it
- Use argparse for command line arguments
- Add basic unit tests
- Update README
- Use ARID for sample unit reference
- Delete the old spec example sample files

## How to Test
Build and run the image against an rm cluster or docker dev/the docker compose backing services. The sample loader should still deliver all case messages onto the case delivery queue and sample attributes into Redis. Note that these cases and attributes will not be valid and may cause errors in case/action/actionexporter until the other changes required for this ticket are made.

## Links
https://trello.com/c/e9iWXgje/517-rmc-83-load-hh-sample-and-generate-print-file-to-census-spec-13
https://github.com/ONSdigital/census-rm-actionexporter-service/pull/3
https://github.com/ONSdigital/census-rm-acceptance-tests/pull/34
https://github.com/ONSdigital/census-rm-action-service/pull/3
